### PR TITLE
Fix Query Frontend grpc settings to avoid noisy error log

### LINKED ISSUE #493: Noisy error log in frontend processor - "transport is closing"
**Tempo backend not ready to receive traffic even after hours**

Probably due to the following
````
level=error ts=2021-01-30T15:47:14.273319231Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
````

**Full Log**
````
level=info ts=2021-01-30T15:45:53.905814194Z caller=main.go:89 msg="Starting Tempo" version="(version=c189e23e, branch=master, revision=c189e23e)"
--
  | level=info ts=2021-01-30T15:45:54.258823037Z caller=server.go:229 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
  | level=info ts=2021-01-30T15:45:54.260118522Z caller=frontend.go:24 msg="creating tripperware in query frontend to shard queries"
  | level=warn ts=2021-01-30T15:45:54.260443879Z caller=modules.go:140 msg="Worker address is empty in single binary mode.  Attempting automatic worker configuration.  If queries are unresponsive consider configuring the worker explicitly." address=127.0.0.1:9095
  | level=info ts=2021-01-30T15:45:54.26054227Z caller=worker.go:112 msg="Starting querier worker connected to query-frontend" frontend=127.0.0.1:9095
  | ts=2021-01-30T15:45:54Z level=info msg="OTel Shim Logger Initialized" component=tempo
  | level=info ts=2021-01-30T15:45:54.261646095Z caller=module_service.go:58 msg=initialising module=memberlist-kv
  | level=info ts=2021-01-30T15:45:54.261675553Z caller=module_service.go:58 msg=initialising module=overrides
  | level=info ts=2021-01-30T15:45:54.261691519Z caller=module_service.go:58 msg=initialising module=store
  | level=info ts=2021-01-30T15:45:54.26172941Z caller=module_service.go:58 msg=initialising module=server
  | level=info ts=2021-01-30T15:45:54.263537314Z caller=module_service.go:58 msg=initialising module=ring
  | level=info ts=2021-01-30T15:45:54.263580354Z caller=module_service.go:58 msg=initialising module=ingester
  | level=info ts=2021-01-30T15:45:54.26361552Z caller=module_service.go:58 msg=initialising module=compactor
  | level=info ts=2021-01-30T15:45:54.263636317Z caller=module_service.go:58 msg=initialising module=query-frontend
  | level=info ts=2021-01-30T15:45:54.26381546Z caller=module_service.go:58 msg=initialising module=querier
  | level=info ts=2021-01-30T15:45:54.263859051Z caller=module_service.go:58 msg=initialising module=distributor
  | level=info ts=2021-01-30T15:45:54.263973679Z caller=worker.go:192 msg="adding connection" addr=127.0.0.1:9095
  | level=info ts=2021-01-30T15:45:54.264012639Z caller=ingester.go:278 msg="beginning wal replay" numBlocks=0
  | level=info ts=2021-01-30T15:45:54.263814539Z caller=compactor.go:95 msg="waiting for compaction ring to settle" waitDuration=1m0s
  | level=info ts=2021-01-30T15:45:54.264449966Z caller=lifecycler.go:521 msg="not loading tokens from file, tokens file path is empty"
  | level=info ts=2021-01-30T15:45:54.26545844Z caller=client.go:242 msg="value is nil" key=collectors/ring index=1
  | level=info ts=2021-01-30T15:45:54.266328848Z caller=lifecycler.go:550 msg="instance not found in ring, adding with no tokens" ring=ingester
  | level=info ts=2021-01-30T15:45:54.266478004Z caller=lifecycler.go:397 msg="auto-joining cluster after timeout" ring=ingester
  | ts=2021-01-30T15:45:54Z level=info msg="No sampling strategies provided, using defaults" component=tempo
  | level=info ts=2021-01-30T15:45:54.267568344Z caller=app.go:212 msg="Tempo started"
  | level=info ts=2021-01-30T15:46:54.265244168Z caller=compactor.go:97 msg="enabling compaction"
  | level=info ts=2021-01-30T15:46:54.265316073Z caller=tempodb.go:278 msg="compaction and retention enabled."
  | level=error ts=2021-01-30T15:47:14.273319231Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
  | level=error ts=2021-01-30T15:47:14.273303551Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
  | level=error ts=2021-01-30T15:47:14.273303003Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
  | level=error ts=2021-01-30T15:47:14.27332587Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
  | level=error ts=2021-01-30T15:47:14.273321788Z caller=frontend_processor.go:61 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"

````

image

**To Reproduce**
1. Start Tempo (grafana/tempo:latest) , as per the blog https://reachmnadeem.wordpress.com/2021/01/30/distributed-tracing-using-grafana-tempo-jaeger-with-amazon-s3-as-backend-in-openshift-kubernetes/

**Expected behavior**

Tempo backend should be ready to receive traffic

**Environment:**
 - Infrastructure: [Kubernetes, Openshift]
 - Deployment tool: [Jenkins Pipeline]

**Additional Context**

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

**What this PR does**:
Fix Query Frontend GRPC settings to fix the noisy error message 

```
caller=frontend_processor.go:59 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
```

Thanks to @pstibrany for digging into this in Cortex - https://github.com/cortexproject/cortex/issues/3606#issuecomment-745179803

**Which issue(s) this PR fixes**:
Fixes #493 

**Checklist**
- NA Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by dgzlopes — APPROVED** [review]

**Comment by joe-elliott** [comment]
Oh dang, nice!

Does this make grpc heartbeat more frequently between querier->queryfrontend so the disconnects don't happen?  if so can we default to these lower values?

**Comment by mdisibio** [comment]
Do we want to set defaults in .go, rather than require most users to set these settings to avoid noisy errors?

**Comment by annanay25** [comment]
Thanks for the comments. Updated the PR to make the changes in the .go file

@joe-elliott - these settings will not affect the heartbeat messages frequency. It is a setting on the server that prevents it from sending a `GOAWAY` message if a client sends heartbeat messages frequently (due to lack of real traffic)

File: cmd/tempo/app/app.go [hunk 1]
```diff
@@ -70,6 +71,8 @@ func (c *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet) {
 	// Server settings
 	flagext.DefaultValues(&c.Server)
 	c.Server.LogLevel.RegisterFlags(f)
+	c.Server.GRPCServerMinTimeBetweenPings = 10 * time.Second
```
Comment (joe-elliott) [inline cmd/tempo/app/app.go:74]: please add a comment as to why this is being set

File: operations/kube-manifests/Namespace-tracing.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (joe-elliott) [inline operations/kube-manifests/Namespace-tracing.yaml:1]: do we need this?

**Comment by pstibrany** [comment]
Here is one way of changing the default, without requiring change in the upstream library: https://github.com/cortexproject/cortex/pull/4168.

File: operations/kube-manifests/Namespace-tracing.yaml (hunk 2 above)
Comment (annanay25) [reply operations/kube-manifests/Namespace-tracing.yaml:1]: Yep, generated by `tk`

**Pull request merged**